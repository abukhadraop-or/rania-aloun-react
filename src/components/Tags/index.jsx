import React, { useState, useEffect } from 'react';
import { getAllTags } from 'services/tags-service';
import ProbTypes from 'prop-types';
import {
  Tag,
  TagButton,
  Container,
  TagsHeader,
  TagsContainer,
} from './tags.styles';

/**
 * Component holding the tags in a container.
 *
 * @param {Object}   props Props passed to Tags component.
 * @param {function} props.onTagSelect Changes previewed articles based on tag selection.
 *
 * @return {JSX.Element} Container holding all the tags of the articles.
 */
function Tags({ onTagSelect }) {
  const [tags, setTags] = useState([]);

  useEffect(() => {
    const fetchTags = async () => {
      const { data } = await getAllTags();
      setTags(data);
    };
    fetchTags();
  }, []);

  return (
    <Container>
      <TagsContainer>
        <TagsHeader>Popular Tags</TagsHeader>
        <TagButton onClick={() => onTagSelect('allArticles')}>
          <Tag>All</Tag>
        </TagButton>
        {tags.map((item) => (
          <TagButton key={item.id} onClick={() => onTagSelect(item.name)}>
            <Tag>{item.name}</Tag>
          </TagButton>
        ))}
      </TagsContainer>
    </Container>
  );
}

Tags.propTypes = {
  onTagSelect: ProbTypes.func.isRequired,
};

export default Tags;
