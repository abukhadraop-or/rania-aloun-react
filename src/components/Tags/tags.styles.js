import styled from 'styled-components';
import colors from 'styles/colors';

export const Container = styled.div`
  margin: 0 1rem;
`;

export const TagsHeader = styled.p`
  font-family: 'source sans pro', sans-serif;
  font-size: 1.3rem;
  margin: 0 0 0.5rem 0;
  padding: 0 0.5rem;
  width: 100%;
`;

export const TagsContainer = styled.div`
  background-color: ${colors.aquaHaze};
  border-radius: 0.125rem;
  display: flex;
  flex-wrap: wrap;
  margin: 1rem 0;
  padding: 0.3rem;
`;

export const TagButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
`;

export const Tag = styled.p`
  background-color: ${colors.osloGrey};
  border-radius: 0.8rem;
  color: #fff;
  cursor: pointer;
  font-size: 0.8rem;
  height: fit-content;
  margin: 0.125rem;
  padding: 0.2rem;
  width: fit-content;

  &:hover {
    background-color: ${colors.smokeyGrey};
  }
`;
