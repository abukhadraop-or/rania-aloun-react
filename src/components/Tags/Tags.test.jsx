import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// import getTags from 'services/tags-service';
// import { act } from '@testing-library/react';
import Tags from '.';
// import { Tag } from './tags.styles';

// jest.mock('services/get-tags', () => jest.fn());

// const mockedTags = {
//   data: [
//     {
//       createdAt: '2021-09-27T03:27:53.000Z',
//       id: 1,
//       name: 'Article',
//       updatedAt: '2021-09-27T03:27:53.000Z',
//     },
//   ],
// };

describe('<Tags />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Tags onTagSelect={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //   it('sets fetched data to tags', async () => {
  //     const promise = Promise.resolve();
  //     getTags.mockReturnValue(mockedTags);
  //     const wrapper = mount(<Tags onTagSelect={jest.fn()} />);
  //     await act(() => promise);
  //     wrapper.update();
  //     const { tags } = wrapper.find(Tag).text();
  //     expect(tags).toEqual(mockedTags.data.name);
  //   });
});
