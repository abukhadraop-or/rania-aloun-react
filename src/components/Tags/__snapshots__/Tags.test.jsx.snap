// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.p>
      Popular Tags
    </styled.p>
    <styled.button
      onClick={[Function]}
    >
      <styled.p>
        All
      </styled.p>
    </styled.button>
  </styled.div>
</styled.div>
`;
